source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,276 @@
+
+
+Using the **Aggregation builders** API, the same operation may look something like:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/agg-factory.java
+   :language: java
+   :copyable: false
+
+Which will produce output similar to the following:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id": 0, "red": 3, "blue": 2}
+
+Useful Links
+~~~~~~~~~~~~
+
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+- :ref:`Aggregation Builders <builders>`
+
+Runnable Examples
+-----------------
+
+The following examples require a ``restaurants`` collection in the ``test`` database. To create and populate the collection,
+follow the `directions <https://github.com/mongodb/docs-assets/tree/drivers>`_ to download and load the dataset. If you
+prefer to load the data directly and have both ``mongoimport`` and ``curl`` in your path, you can run the following instead:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Atlas
+
+      .. code-block:: console
+
+         curl -O https://raw.githubusercontent.com/mongodb/docs-assets/drivers/restaurants.json
+         mongoimport --uri mongodb+srv://<your-atlas-uri>/test --collection restaurants --drop --file restaurants.json
+
+
+
+   .. tab::
+      :tabid: localhost
+
+      .. code-block:: console
+
+         curl -O https://raw.githubusercontent.com/mongodb/docs-assets/drivers/restaurants.json
+         mongoimport -d test -c restaurants --drop --file restaurants.json
+
+Base Setup
+~~~~~~~~~~
+
+Create a new Java file called ``AggTour.java`` and include the following import statements:
+
+.. code-block:: java
+
+   import com.mongodb.client.MongoClient;
+   import com.mongodb.client.MongoClients;
+   import com.mongodb.client.MongoCollection;
+   import com.mongodb.client.MongoDatabase;
+   import com.mongodb.client.model.Accumulators;
+   import com.mongodb.client.model.Aggregates;
+   import com.mongodb.client.model.Filters;
+   import org.bson.Document;
+
+   import java.util.Arrays;
+
+Connect to a MongoDB Deployment
++++++++++++++++++++++++++++++++
+
+Connect to a MongoDB deployment and declare and define ``MongoDatabase`` and ``MongoCollection`` instances.
+
+For example, include the following code to connect to a standalone MongoDB deployment running on localhost on port
+``27017`` and define ``database`` to refer to the ``test`` database and ``collection`` to refer to the ``restaurants``
+collection.
+
+.. code-block:: java
+
+   public class AggTour {
+
+       public static void main(String[] args) {
+           MongoClient mongoClient = MongoClients.create();
+           MongoDatabase database = mongoClient.getDatabase("test");
+           MongoCollection<Document> collection = database.getCollection("restaurants");
+
+           // aggregation here
```

Comment:
This was confusing to me initially, but think i figured it out by looking at the example. To make this a little more clear, I might say "// placeholder for your aggregation" and then either "// your connection setup code" or omit completely in the example on line 193.

Reply:
Okay, I can see how there can be disconnect.